docs/citations.rst

```diff
@@ -24,10 +24,10 @@ Updating list of citations and use cases of :code:`pyhf`:
    :all:
    :style: plain
 
-Published Likelihoods
----------------------
+Published Probability Models
```

Comment:
I could go either way with "probability models" or "statistical models".

Reply:
statistical model. it's the more common phrasing I think.